src/idl_parser.cpp

```diff
@@ -897,6 +897,36 @@ CheckedError Parser::ParseTable(const StructDef &struct_def, std::string *value,
     EXPECT(',');
   }
 
+  // Check if all required fields are parsed.
+  std::vector<FieldDef*> requiredFields;
+  auto rfIt = requiredFields.begin();
+  for (auto fieldIt = struct_def.fields.vec.begin();
```

Comment:
This entire first loop (and vector) are not necessary. Just make this iteration part of the second loop.